apps/rpc/src/modules/workspace-sync/client.ts

```diff
@@ -0,0 +1,58 @@
+import { type admin_directory_v1, google } from "googleapis"
+import z from "zod"
+import { isSyncEnabled } from "./helpers"
+
+const SCOPES = [
+  "https://www.googleapis.com/auth/admin.directory.group",
+  "https://www.googleapis.com/auth/admin.directory.group.member",
+  "https://www.googleapis.com/auth/admin.directory.user",
+  "https://www.googleapis.com/auth/admin.directory.user.alias",
+  "https://www.googleapis.com/auth/admin.directory.user.security",
+]
+
+const serviceAccountJsonSchema = z.object({
+  auth_provider_x509_cert_url: z.string().url(),
+  auth_uri: z.string().url(),
+  client_email: z.string().email(),
+  client_id: z.string().min(1).max(100),
+  client_x509_cert_url: z.string().url(),
+  private_key: z.string().min(1).max(1000),
+  private_key_id: z.string().min(1).max(100),
+  project_id: z.string().min(1).max(100),
+  token_uri: z.string().url(),
+  type: z.literal("service_account"),
+})
+
+const getJWT = () => {
```

Comment:
Pretty sure you can authenticate against google apis using just the service account, and not forging a jwt